io/src/main/scala/fs2/io/ToInputStream.scala

```diff
@@ -0,0 +1,259 @@
+package fs2.io
+
+
+import java.io.{IOException, InputStream}
+
+
+import fs2._
+import Chunk.Bytes
+import util.{Async, Attempt}
+import async.mutable
+import Async.Change
+
+import scala.concurrent.SyncVar
+
+
+/**
+  * Conversion of Stream[F,Byte] to java.io.InputStream
+  */
+object ToInputStream {
```

Comment:
I think we should put this code inside the `JavaInputOutputStream` object instead of its own object. To hide the implementation methods, just nest them inside the main combinator. E.g.,

```scala
private[io] object JavaInputOutputStream {
  ...
  def toInputStream[F[_]](implicit F: Async[F]): Pipe[F,Byte,InputStream] = {
    // Put everything here, including the contents of the impl package
  }
}
```